src/mlpack/methods/bayesian_ridge/bayesian_ridge.hpp

```diff
@@ -0,0 +1,320 @@
+ 
+ * // Train the model. Regularization strength is optimally tunned with the
+ * // training data alone by applying the Train method.
+ * BayesianRidge estimator(); // Instanciate the estimator with default option.
+ * estimator.Train(Xtrain, ytrain);
+ 
+ * // Prediction on test points.
+ * arma::mat Xtest; // Test data matrix. Column-major.
+ * arma::rowvec predictions;
+ 
+ * estimator.Predict(Xtest, prediction);
+ 
+ * arma::rowvec ytest; // Test target values.
+ * estimator.Rmse(Xtest, ytest); // Evaluate using the RMSE score.
+ 
+ * // Compute the standard deviations of the predictions.
+ * arma::rowvec stds;
+ * estimator.Predict(Xtest, responses, stds)
+ * @endcode
+ */
+class BayesianRidge
+{
+ public:
+  /**
+   * Set the parameters of Bayesian Ridge regression object. The
+   *    regulariation parameter is automaticaly set to its optimal value by
+   *    maximmization of the marginal likelihood.
+   *
+   * @param fitIntercept Whether or not center the data according to the
+   *    examples.
+   * @param normalize Whether or to normalize the data according to the
+   *    standard deviation of each feature.
+   **/
+  BayesianRidge(const bool fitIntercept = true,
+                const bool normalize = false);
+
+  /**
+   * Run BayesianRidge regression. The input matrix (like all mlpack matrices)
+   * should be
+   * column-major -- each column is an observation and each row is a dimension.
+   * 
+   * @param data Column-major input data
+   * @param responses A vector of targets.
+   * @return score. Root Mean Square Error. Equal to -1 of two feature vectors 
+   *    or more are colinear.
+   **/
+  float Train(const arma::mat& data,
+              const arma::rowvec& responses);
+
+  /**
+   * Predict \f$y_{i}\f$ for each data point in the given data matrix using the
+   * currently-trained Bayesian Ridge model.
+   *
+   * @param points The data points to apply the model.
+   * @param predictions y, Contain the  predicted values on completion.
+   **/
+  void Predict(const arma::mat& points,
+               arma::rowvec& predictions) const;
+
+  /**
+   * Predict \f$y_{i}\f$ for one point using the
+   * currently trained Bayesian Ridge model.
+   *
+   * @param point The data point to apply the model.
+   * @param prediction y, which will contained predicted value on completion.
+   **/
+
+  void Predict(const arma::colvec& point, double& prediction) const;
+
+  /**
+   * Predict \f$y_{i}\f$ and the standard deviation of the predictive posterior 
+   * distribution for each data point in the given data matrix, using the
+   * currently-trained Bayesian Ridge estimator.
+   *
+   * @param points The data point to apply the model.
+   * @param predictions y, which will contained calculated values on completion.
+   * @param std Standard deviations of the predictions.
+   */
+  void Predict(const arma::mat& points,
+               arma::rowvec& predictions,
+               arma::rowvec& std) const;
+
```

Comment:
Another pedantic style comment; looks like we can remove the empty line here. There are some other places where this applies as well.